share/revng/test/configuration/revng-qa/runtime-abi-test-toolchain.yml

```diff
@@ -0,0 +1,69 @@
+tags:
+  - name: abi-test-sources
+  - name: abi-test-asm-sources
+    implies: [abi-test-sources]
+  - name: abi-test-c-sources
+    implies: [abi-test-sources]
+    variables:
+      CFLAGS:
+        - -std=c99
+        - -fno-pic
+        - -fno-pie
+        - -fno-stack-protector
+        - -fno-zero-initialized-in-bss
+        - -O1
+
+  - name: abi-test-argument-setup
+    implies: [abi-test-asm-sources]
+  - name: abi-test-return-value-setup
+    implies: [abi-test-c-sources]
+
+  - name: abi-test-function-prototypes
+    implies: [simple-executable]
+    variables:
+      CFLAGS:
+        - -std=c99
+        - -fno-pic
+        - -fno-pie
+        - -fno-stack-protector
+        - -ggdb3
+        - -O3
+
+  - name: abi-test-symbol-gathering-script
+  - name: abi-test-list
+  - name: abi-test-printers
+  - name: abi-test-runner
+
+  - name: for-abi-runtime
+
+commands:
+  - type: revng-qa.compiled-to-object-file
+    from:
+      - type: source
+        filter: abi-test-sources and linux
+    suffix: .o
+    command: ${TRIPLE}gcc ${INPUT} -o ${OUTPUT} ${CFLAGS} -c
+
+  # I'd rather I didn't have to duplicate it, but it causes a dependency cycle otherwise
+  - type: revng-qa.compiled-with-debug-info-copy
+    from:
+      - type: revng-qa.compiled-to-object-file
+        filter: abi-test-return-value-setup
+      - type: revng-qa.compiled-to-object-file
+        filter: abi-test-argument-setup
+    command: ${TRIPLE}gcc ${INPUT1} ${INPUT2} -o ${OUTPUT} ${CFLAGS} -Wl,--section-start=.text=0x2000000
+
+  - type: revng-qa.compiled-with-debug-info
+    from:
+      - type: revng-qa.compiled-with-debug-info-copy
+      - type: source
+        filter: abi-test-runner
+      - type: source
+        filter: abi-test-printers
+    command: |-
+      GENERATED_HEADERS=$$(temp -d);
+      ${TRIPLE}objdump --wide --no-show-raw-insn ${OBJDUMP_FLAGS} -h -t -d ${INPUT1}
```

Comment:
(since the topic has no thread (we discussed it in Mattermost), I'll open one. cc: @aleclearmind)

You asked for simplification of the structure of these `command`s, what do you think about something like this?

Reply:
Still way way too complex, several points I mentioned are not addressed.

1. `test_list.txt` can go too. All of its content is deterministic (see [this comment](https://github.com/revng/revng-qa/pull/17#discussion_r1354348110)).
2. Reorganize all the files as follows:
   1. Only one entry in `sources`, targeting a *directory*, `share/revng/test/tests/abi/`. We never did this, if it doesn't work, let's fix `revng-test-configure` so that it works.
        This entry must have `repeat-for:` all the ABIs.
   2. Only *one* command that takes *one* input, the source above and does *everything*. From the only input you can compute the path to all the source files you want.

   This should make go away many tags.

I'd like to stress that commands whose results are used as input in another command are not OK.

Reply:
I left the `simple-executable` one intact, if that's ok with you. Other than that, please take a look.